src/app/pages/details/common/hooks.tsx

```diff
@@ -3,28 +3,26 @@ import tableOfContentsHtml from '~/models/table-of-contents-html';
 import partnerFeaturePromise, {tooltipText} from '~/models/salesforce-partners';
 import shuffle from 'lodash/shuffle';
 import {camelCaseKeys} from '~/helpers/page-data-utils';
+import useDetailsContext from '../context';
 
-export function useTableOfContents(model) {
-    const webviewLink = model.webviewRexLink || model.webviewLink;
-    const isRex = Boolean(model.webviewRexLink);
-    const [tocHtml, setTocHtml] = useState('');
+export function useTableOfContents() {
+    const model = useDetailsContext();
+    const webviewLink = model.webviewRexLink;
+    const [tocHtml, setTocHtml] = useState<string | void>('');
 
     tableOfContentsHtml({
-        isRex,
         cnxId: model.cnxId,
-        bookSlug: model.slug,
         webviewLink
-    }).then(
-        setTocHtml,
-        (err) => {
-            console.warn(`Failed to generate table of contents HTML for ${model.cnxId}: ${err}`);
-        }
-    );
+    }).then(setTocHtml, (err) => {
+        console.warn(
+            `Failed to generate table of contents HTML for ${model.cnxId}: ${err}`
+        );
+    });
 
-    return tocHtml;
+    return tocHtml as string;
 }
 
-function toBlurb(partner) {
+function toBlurb(partner: object) {
```

Comment:
Can we make the type for this? Or are you waiting to convert the model first?

Reply:
Didn't realize I already had a type for it.